cobbler/modules/managers/import_signatures.py

```diff
@@ -545,7 +545,7 @@ def arch_walker(self, foo: dict, dirname: str, fnames: list):
                     re_krn2 = re.compile(self.signature["kernel_arch_regex"])
                     krn_lines = self.get_file_lines(os.path.join(dirname, x))
                     for line in krn_lines:
-                        m = re_krn2.match(line)
+                        m = re_krn2.match(line.decode("utf-8", "ignore"))
```

Comment:
Is ignoring byte decoding errors the best approach or should we use `"replace"` or even `"strict"`?  What encodings do we expect in these files?

Reply:
If you look at the function above we handle multiple cases. Since those magic file reading logic may not handle all cases correctly, I don't know the answer. It could be that the data present at that point is wrong and thus `get_file_lines()` isn't working correctly.

For me personally the way forward is: Investigate the way `cobbler import` is working in relation to the types and then take an informed decision.

Reply:
What type is `get_file_lines` supposed to return?

Reply:
I checked `get_file_lines`, it seems just return readed file content.

Reply:
So I investigated. The story has three cases:

1. [Line 126](https://github.com/cobbler/cobbler/blob/e36be05e3792c28b114fb9e3855a3aa90387e6be/cobbler/modules/managers/import_signatures.py#L126): The problematic case where at the moment the preferred encoding is taken. (See [Source](https://docs.python.org/3.8/library/io.html#io.TextIOWrapper) because we do specify `r` for reading and not `rb`.)
2. [Line 134](https://github.com/cobbler/cobbler/blob/e36be05e3792c28b114fb9e3855a3aa90387e6be/cobbler/modules/managers/import_signatures.py#L134): Unproblematic because gives us `str`
3. [Line 139](https://github.com/cobbler/cobbler/blob/e36be05e3792c28b114fb9e3855a3aa90387e6be/cobbler/modules/managers/import_signatures.py#L139): Unproblematic because gives us `str`

@agraul I see no "easy" way because we don't know what encoding the file has. Do you have any idea?

Reply:
Okay thanks for the update. I will think more once I have time how to fix this. Is it a problem for you to continue to leave this open?

Reply:
Not at all. You can just leave it open until you or others found solution. :)  Thank you.